### ISSUE TITLE
Remove benchmark feature in stable builds

### ISSUE THREAD

**Issue by jdm** [servo/rust-smallvec#44]
From travis:
```
     Running `rustc --crate-name bench benches/bench.rs --emit=dep-info,link -C opt-level=3 --test -C metadata=6041378d75069d53 -C extra-filename=-6041378d75069d53 --out-dir /home/travis/build/servo/rust-smallvec/target/release/deps -L dependency=/home/travis/build/servo/rust-smallvec/target/release/deps --extern smallvec=/home/travis/build/servo/rust-smallvec/target/release/deps/libsmallvec-ad2736f846965482.rlib`

error[E0554]: #[feature] may not be used on the stable release channel

 --> benches/bench.rs:1:1

  |

1 | #![feature(test)]

  | ^^^^^^^^^^^^^^^^^

error: aborting due to previous error

error: Could not compile `smallvec`.

Caused by:

  process didn't exit successfully: `rustc --crate-name bench benches/bench.rs --emit=dep-info,link -C opt-level=3 --test -C metadata=6041378d75069d53 -C extra-filename=-6041378d75069d53 --out-dir /home/travis/build/servo/rust-smallvec/target/release/deps -L dependency=/home/travis/build/servo/rust-smallvec/target/release/deps --extern smallvec=/home/travis/build/servo/rust-smallvec/target/release/deps/libsmallvec-ad2736f846965482.rlib` (exit code: 101)

Build failed, waiting for other jobs to finish...

error: build failed

The command "cargo build --verbose &&

cargo build --features=heapsizeof --verbose &&

cargo test --verbose &&

cargo test --features=heapsizeof --verbose &&

([ $TRAVIS_RUST_VERSION != nightly ] || cargo bench --verbose bench)

" exited with 101.
```

**Comment by mbrubeck** [comment]
The travis script should already do this, but its check seems to be broken, maybe related to travis-ci/travis-build#993 which just landed.

**Comment by mbrubeck** [comment]
I temporarily disabled `cargo bench`.  We should re-enable it after the Travis issue is fixed.

**Comment by mbrubeck** [comment]
Fixed by travis-ci/travis-ci#7510.
